src/Dominio/Persistencia.java

```diff
@@ -0,0 +1,18 @@
+package Dominio;
+
+import java.util.ArrayList;
+
+public class Persistencia {
+    private GestorDeSocios gestor;
+
+    public Persistencia() {
+        this.gestor = gestor;
```

Comment:
Aquí el constructor no recibe ningún argumento, y la asignación es innecesaria porque le asignamos lo mismo. `this.gestor` es exactamente lo mismo que `gestor`